### PR TITLE
[DCJ-3] Enable Dependabot for Gradle updates

### PR DESCRIPTION
**Pull request by okotsopoulos**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/DCJ-3

## Addresses

Presently, we update dependencies in TDR on a "just in time" basis:
- As needed for development
- To upgrade past an end-of-life version
- When required to pull in a security patch

But these dependency jumps can often be quite large and make for difficult upgrade experiences.

The standard throughout DSP has been to use Dependabot to automate dependency upgrades (which has also shown the benefit of mitigating security vulnerabilities before they're flagged without additional out-of-band-effort).  Now that all of the repository secrets used in PR workflows are managed by Terraform and exposed to Dependabot (see https://github.com/broadinstitute/terraform-ap-deployments/pull/1668) we can enable Dependabot in this repository as well.

## Summary of changes

- Use Dependabot to automate dependency upgrades in this repository, with past experience working on [WSM](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/.github/dependabot.yml) as a guide
- Minor and patch upgrades will be grouped together, major upgrades will be PRed individually as they are more likely to contain breaking changes

## Testing Strategy

None: we will iterate on this workflow if needed.

### CODE REVIEW

**Review by rjohanek — APPROVED** [review]
this looks great. super excited about this!

**Review by snf2ye — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        exclude-patterns:
+          - "spotless-plugin-gradle" # likely to require reformatting of code
+        update-types:
+          - "minor"
+          - "patch"
+    schedule:
+      interval: "weekly"
+      time: "06:00"
+      timezone: "America/New_York"
+    target-branch: "main"
```
Comment (pshapiro4broad) [inline .github/dependabot.yml:29]: Is this correct? I would expect it to be `develop`

Comment (okotsopoulos) [reply .github/dependabot.yml:29]: Thank you for catching this copypasta -- I'll PR a fix.

Comment (okotsopoulos) [reply .github/dependabot.yml:29]: https://github.com/DataBiosphere/jade-data-repo/pull/1762

**Pull request merged**